torchmetrics/detection/mean_ap.py

```diff
@@ -482,9 +482,11 @@ def __evaluate_image_gt_no_preds(
     ) -> Dict[str, Any]:
         """Some GT but no predictions."""
         # GTs
-        gt = gt[gt_label_mask]
+        gt = np.asarray(gt)[gt_label_mask]
+        if len(gt_label_mask) == 1:
+            gt = np.expand_dims(gt, axis=0)
```

Comment:
Shouldn't we use directly `torch` instead of `numpy`?

Reply:
`gt` is a list like [(width, height), b"some_RLE_encoding"]. the binary part is not torch encodable + the input is later on given to pycocotools, which assumes np/python.

Happy to learn how to convert things to torch if you give an example.

Reply:
Oh, was not aware of this. In that case, sticking to `numpy` seems to fine to me :]

Reply:
Turns out it was still breaking some bbox assumptions about torch.stack (it had to be np dtype=object), so final solution that I feel is actually cleaner is to do list comprehension (avoids some if statements)